pandas/core/frame.py

```diff
@@ -5318,7 +5325,7 @@ def _arith_op(left, right):
 
     def _combine_match_index(self, other, func, level=None):
         left, right = self.align(other, join="outer", axis=0, level=level, copy=False)
-        assert left.index.equals(right.index)
+        # at this point we have `left.index.equals(right.index)`
```

Comment:
Why do we want to remove the asserts here? If someone is that set on getting this optimization couldn't they just use the -OO flag?

Reply:
i put these assertions in when trying to understand the ops code a while back.  having a comment to the same effect performs the same task for the reader.